### DISCUSSION TITLE
Multiple input files for single sub-app

### DISCUSSION THREAD

**Discussion by maxposchmann** [idaholab/moose#23762] [Q&A Modules: General] [answered]
Sometimes it is handy to collect common inputs in one input file, then add on some other stuff in another input file. On the command line, I frequently use syntax like
```
./moose-opt -i common_input.i additional_input.i
```
I'd like to be able to use similar syntax to run a sub-app (in particular using stochastic tools with `SamplerFullSolveMultiApp`), but entering multiple input files in the `input_files` parameter does something else entirely.

I thought maybe I could use `cli_args` to do what I wanted like:
```
[MultiApps]
  [sub]
    type = SamplerFullSolveMultiApp
    sampler = sampler
    input_files = 'common_input.i'
    cli_args = '-i additional_input.i'
  []
[]
```
However, that doesn't get parsed the way I want, and MOOSE complains: `The number of items supplied must be 1 or equal to the number of sub apps.`

Am I missing something that will make one of these options work? Is there another option?

**Comment by GiudGiud** [comment]
Hello

I dont think we've made that work inside MultiApps. But can you try this? without the -i
```
[MultiApps]
  [sub]
    type = SamplerFullSolveMultiApp
    sampler = sampler
    input_files = 'common_input.i'
    cli_args = 'additional_input.i'
  []
[]
```
We'd be interested in the contribution if not

Guillaume

**Reply by maxposchmann** [reply]
Thanks for the suggestion. When I tried that, I got `Number of command line arguments does not match number of sampler columns.`

**Reply by maxposchmann** [reply]
I'm not sure what to suggest as the "correct" way to handle this. Maybe `input_files` could support some sort of grouping (either like `{}` or separated by `;`). This can easily get messy since it is already a vector.

Another thought is that a way to achieve equivalent (but not identical) behavior would be to use input file include directives as mentioned in #20125 and #9755. Looking at those threads, it's not clear if that will eventually be implemented or not.

**Reply by maxposchmann** [reply]
For other readers, in [MooseApp.C](https://github.com/idaholab/moose/blob/17ca2808a0aee4e1eb8c0ee39d3c9decb34f5a53/framework/src/base/MooseApp.C#L1022) there is a comment:
`for now we only permit single input to be set for multiapps`

**Reply by GiudGiud** [reply, marked as answer]
The logic handling this parameter just needs to be adapted
```
    input_files = 'common_input.i'
```
Right now we consider 1 input per app, but we'd need some sort of vector syntax indeed
